### PR TITLE
Update PGCGAP to v1.0.35

### PR DESCRIPTION
**Pull request by liaochenlanruo**
Remove Roary and use Panaroo instead.
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by liaochenlanruo** [comment]
@BiocondaBot please fetch artifacts

**Comment by liaochenlanruo** [comment]
@BiocondaBot please fetch artifacts

**Comment by liaochenlanruo** [comment]
@BiocondaBot please add label

**Comment by liaochenlanruo** [comment]
Thank you @bgruening  for the review of #33794, but that is the auto bump one, and this one was the new version I want to submit and merge.

File: recipes/pgcgap/meta.yaml
Comment (dpryan79) [inline recipes/pgcgap/meta.yaml:19]: This will make it incompatible with everything else in bioconda. Are you sure you want to do this?

**Review by dpryan79 — CHANGES_REQUESTED** [review]
We should be careful about pinning this to old perl versions.

**Comment by dpryan79** [comment]
Maybe it's simplest to wait until I've merged the changes to bioconda-utils that use boa/mambabuild.

File: recipes/pgcgap/meta.yaml
Comment (liaochenlanruo) [reply recipes/pgcgap/meta.yaml]: Thanks for your suggestion, I had removed the version of Perl and started a new test to see if it could pass the test.  One more thing, I wonder when can we use the mambabuild.  I am very expecting the new builder because it is painful to use the old one for recipes with a lot of dependency packages.

Comment (liaochenlanruo) [reply recipes/pgcgap/meta.yaml]: Thanks for your suggestion, I had removed the version of Perl and started a new test to see if it could pass the test.  One more thing, I wonder when can we use the mambabuild.  I am very expecting the new builder because it is painful to use the old one for recipes with a lot of dependency packages.

Comment (dpryan79) [reply recipes/pgcgap/meta.yaml]: I've just switched over to mambabuild and will use this as the test PR to ensure that's fully working :)

**Comment by dpryan79** [comment]
Ah, roary needs to be rebuilt (I assume it's blacklisted). BTW, I need to temporarily switch back to the older bioconda-utils that's not using mambabuild. There's some dependency conflict on OSX.

File: recipes/pgcgap/meta.yaml
Comment (liaochenlanruo) [reply recipes/pgcgap/meta.yaml]: Thanks for your kind help! I am honored that this PR can be used for mambabuild testing and hope it will be successful.

**Comment by dpryan79** [comment]
@BiocondaBot please update

**Comment by liaochenlanruo** [comment]
> @BiocondaBot please update

Will it work if Roary was pined to >3.12.0, which supports BioPerl > 1.7? @dpryan79 

```
roary 3.12.0 pl5.22.0_0
-----------------------
file name   : roary-3.12.0-pl5.22.0_0.tar.bz2
name        : roary
version     : 3.12.0
build       : pl5.22.0_0
build number: 0
size        : 85 KB
license     : GPL-3.0
subdir      : linux-64
url         : https://conda.anaconda.org/bioconda/linux-64/roary-3.12.0-pl5.22.0_0.tar.bz2
md5         : d97cf9beb4b79fec41bdd021bae46a86
dependencies:
  - bedtools
  - blast
  - cd-hit
  - fasttree
  - libgcc
  - mafft
  - mcl
  - parallel
  - perl 5.22.0*
  - perl-array-utils
  - perl-bioperl <1.7
  - perl-bioperl-core <1.7
  - perl-digest-md5-file
  - perl-env-path
  - perl-exception-class
  - perl-file-find-rule
  - perl-file-grep
  - perl-file-slurper
  - perl-graph-readwrite
  - perl-log-log4perl
  - perl-moose
  - perl-perlio-utf8_strict
  - perl-test-files
  - perl-test-most
  - perl-test-output
  - perl-text-csv
  - prank


roary 3.12.0 pl526h470a237_1
----------------------------
file name   : roary-3.12.0-pl526h470a237_1.tar.bz2
name        : roary
version     : 3.12.0
build       : pl526h470a237_1
build number: 1
size        : 87 KB
license     : GPL-3.0
subdir      : linux-64
url         : https://conda.anaconda.org/bioconda/linux-64/roary-3.12.0-pl526h470a237_1.tar.bz2
md5         : c1ca19278083487ac076affdd7cb9468
timestamp   : 2018-09-19 07:06:08 UTC
dependencies:
  - bedtools
  - blast
  - cd-hit
  - fasttree
  - libgcc-ng >=4.9
  - mafft
  - mcl
  - parallel >=20180522
  - perl >=5.26.2,<5.26.3.0a0
  - perl-array-utils
  - perl-bioperl >=1.7.2
  - perl-digest-md5-file
  - perl-exception-class
  - perl-file-find-rule
  - perl-file-grep
  - perl-file-path
  - perl-file-slurper
  - perl-file-temp
  - perl-file-which
  - perl-getopt-long
  - perl-graph
  - perl-graph-readwrite
  - perl-log-log4perl
  - perl-moose
  - perl-perlio-utf8_strict
  - perl-text-csv
  - prank


roary 3.12.0 pl526h516909a_2
----------------------------
file name   : roary-3.12.0-pl526h516909a_2.tar.bz2
name        : roary
version     : 3.12.0
build       : pl526h516909a_2
build number: 2
size        : 88 KB
license     : GPL-3.0
subdir      : linux-64
url         : https://conda.anaconda.org/bioconda/linux-64/roary-3.12.0-pl526h516909a_2.tar.bz2
md5         : f8844b1bb6e9187ccd12caed00d7276e
timestamp   : 2019-10-26 17:55:24 UTC
dependencies:
  - bedtools
  - blast
  - cd-hit
  - fasttree
  - libgcc-ng >=7.3.0
  - mafft
  - mcl
  - parallel >=20180522
  - perl >=5.26.2,<5.26.3.0a0
  - perl-array-utils
  - perl-bioperl >=1.7.2
  - perl-digest-md5-file
  - perl-exception-class
  - perl-file-find-rule
  - perl-file-grep
  - perl-file-path
  - perl-file-slurper
  - perl-file-temp
  - perl-file-which
  - perl-getopt-long
  - perl-graph
  - perl-graph-readwrite
  - perl-log-log4perl
  - perl-moose
  - perl-perlio-utf8_strict
  - perl-text-csv
  - prank


roary 3.13.0 pl526h516909a_0
----------------------------
file name   : roary-3.13.0-pl526h516909a_0.tar.bz2
name        : roary
version     : 3.13.0
build       : pl526h516909a_0
build number: 0
size        : 88 KB
license     : GPL-3.0
subdir      : linux-64
url         : https://conda.anaconda.org/bioconda/linux-64/roary-3.13.0-pl526h516909a_0.tar.bz2
md5         : bae40d1ffb5e60af39589042cc4f2108
timestamp   : 2019-11-07 03:01:41 UTC
dependencies:
  - bedtools
  - blast
  - cd-hit
  - fasttree
  - libgcc-ng >=7.3.0
  - mafft
  - mcl
  - parallel >=20180522
  - perl >=5.26.2,<5.26.3.0a0
  - perl-array-utils
  - perl-bioperl >=1.7.2
  - perl-digest-md5-file
  - perl-exception-class
  - perl-file-find-rule
  - perl-file-grep
  - perl-file-path
  - perl-file-slurper
  - perl-file-temp
  - perl-file-which
  - perl-getopt-long
  - perl-graph
  - perl-graph-readwrite
  - perl-log-log4perl
  - perl-moose
  - perl-perlio-utf8_strict
  - perl-text-csv
  - prank
  ```

**Comment by dpryan79** [comment]
Not until we get roary rebuilt.

**Comment by dpryan79** [comment]
Oh, roary is unmaintained, that's going to be a problem.

**Comment by dpryan79** [comment]
I suspect that most of the roary failures are OK. For example the graph stuff is just a digraph output changing to a graph type, probably due to a syntax change in some dependency. The sequence mismatches are either in stop codons or NNNN stretches, which are presumably also due to parameter changes. Since PGCGAP uses this package, please look into this. The roary `make test` can be commented out of these failing tests can be ignored.

**Comment by liaochenlanruo** [comment]
> I suspect that most of the roary failures are OK. For example the graph stuff is just a digraph output changing to a graph type, probably due to a syntax change in some dependency. The sequence mismatches are either in stop codons or NNNN stretches, which are presumably also due to parameter changes. Since PGCGAP uses this package, please look into this. The roary `make test` can be commented out of these failing tests can be ignored.

Since Roary is unmaintained, I will replace it with the new pipeline [Panaroo](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-02090-4)  which is compatible with other dependency software in the next version of PGCGAP.

**Comment by liaochenlanruo** [comment]
@dpryan79 Why the error code 137 still exists, haven‘t we changed condabuild to mambabuild.?

**Comment by liaochenlanruo** [comment]
@BiocondaBot please fetch artifacts

**Comment by liaochenlanruo** [comment]
@dpryan79 The test has been successed after I specified the version number of most dependicies, please have a review, Thanks!

**Comment by liaochenlanruo** [comment]
@BiocondaBot please add label

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
@BiocondaBot please merge

**Pull request merged**
